### PR TITLE
openapi(dm): add task config template releated openapi

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve? 

subtask of https://github.com/pingcap/ticdc/issues/3583

### What is changed and how it works?

as title, need https://github.com/pingcap/ticdc/pull/3620 merged first

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - Possible performance regression
 - Increased code complexity

Related changes

 - Need to update the documentation

### Release note 

```release-note
`add task config releated openapi`.
```

### CODE REVIEW

File: dm/dm/master/openapi.go [hunk 1]
```diff
@@ -822,6 +823,95 @@ func (s *Server) DMAPIOperateTableStructure(c *gin.Context, taskName string, sou
...
+		return
+	}
+	// prepare target db config
+	newCtx := c.Request.Context()
+	toDBCfg := config.GetTargetDBCfgFromOpenAPITask(task)
+	if adjustDBErr := adjustTargetDB(newCtx, toDBCfg); adjustDBErr != nil {
+		_ = c.Error(terror.WithClass(adjustDBErr, terror.ClassDMMaster))
+		return
+	}
+	if err := ha.PutOpenAPITaskConfig(s.etcdClient, *task, req.Overwrite); err != nil {
```
Comment (glorv) [inline dm/dm/master/openapi.go:845]: So, does this mean we support change all the config fields without any restriction?🤔

**Comment by Ehco1996** [comment]
/hold

need add new batch import api after meeting with PM😅

File: dm/dm/master/openapi.go [hunk 2]
```diff
@@ -822,6 +823,123 @@ func (s *Server) DMAPIOperateTableStructure(c *gin.Context, taskName string, sou
 	}
 }
 
+// DMAPIBatchImportTaskConfig create task_config_template url is: (POST /api/v1/task_configs/batch_import).
+func (s *Server) DMAPIBatchImportTaskConfig(c *gin.Context) {
```
Comment (Ehco1996) [inline dm/dm/master/openapi.go:827]: this api is for this [[feature ](](https://docs.google.com/document/d/1OT70m8rVdMILETZLz0hdpIftP2Mo_jvH66mY9VAtj-E/edit?pli=1&disco=AAAAQ_UCKv4)url)

File: dm/openapi/spec/dm.yaml [hunk 3]
```diff
@@ -809,6 +811,151 @@ paths:
...
+              schema:
+                $ref: "#/components/schemas/BatchImportTaskConfigResponse"
+        "400":
+          description: "failed"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/ErrorWithMessage"
+
+  /api/v1/task_configs:
```
Comment (Ehco1996) [inline dm/openapi/spec/dm.yaml:840]: maybe  use this path pattern ? 🤔

* list/create `api/vi/tasks/configs`?
* batch_import `api/vi/tasks/configs/batch_import`
* delete/get/update `api/vi/tasks/{task-name}/configs` 

need suggestion for other reviewers

**Comment by Ehco1996** [comment]
/unhold

could you please take another look for this pr ? thanks 🧡🧡🧡

/cc @glorv  @okJiang @lance6716

**Comment by Ehco1996** [comment]
please check this api spec of task config  if the api is sufficient for webui

/cc @zoubingwu

File: dm/openapi/spec/dm.yaml [hunk 4]
```diff
@@ -1617,3 +1765,35 @@ components:
...
+          description: whether to overwrite task config template
+      required:
+        - "overwrite"
+    BatchImportTaskConfigResponse:
+      type: object
+      properties:
+        success_task_list:
+          type: array
+          items:
+            type: string
```
Comment (okJiang) [inline dm/openapi/spec/dm.yaml:1784]: How about return the success_task_config_list, not only the task_name

**Review by okJiang — COMMENTED** [review]
rest lgtm

File: dm/openapi/spec/dm.yaml (hunk 4 above)
Comment (Ehco1996) [reply dm/openapi/spec/dm.yaml:1784]: seems webui  don't need to see taskconfigs, actually it only care about the failed msg 😅
you can see the Product prototype drawings [here](https://docs.google.com/document/u/2/d/1OT70m8rVdMILETZLz0hdpIftP2Mo_jvH66mY9VAtj-E/edit?pli=1&disco=AAAAQ_UCKv4)

**Review by okJiang — APPROVED** [review]

**Comment by Ehco1996** [comment]
could you please take another look for this pr ? thanks 🧡🧡🧡 

@glorv  @lichunzhu  @GMHDBJD

File: dm/openapi/spec/dm.yaml (hunk 3 above)
Comment (lichunzhu) [reply dm/openapi/spec/dm.yaml:840]: My point:
list/create api/vi/tasks/configs?
batch_import api/vi/tasks/configs/import
delete/get/update api/vi/tasks/configs/{task-name}

File: dm/openapi/spec/dm.yaml [hunk 5]
```diff
@@ -1617,3 +1765,35 @@ components:
       required:
         - "total"
         - "data"
+
+    BatchImportTaskConfigRequest:
```
Comment (lichunzhu) [inline dm/openapi/spec/dm.yaml:1769]: In dmctl it's called as `config import`. I don't think we should change the name here.
```suggestion
    ImportTaskConfigRequest:
```

**Review by lichunzhu — COMMENTED** [review]
Rest LGTM

File: dm/openapi/spec/dm.yaml (hunk 3 above)
Comment (Ehco1996) [reply dm/openapi/spec/dm.yaml]: addressed in https://github.com/pingcap/tiflow/pull/3656/commits/f585588ef4a73b89f094349bb74229e33cc4d247 
ptal

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**